device/celestica/x86_64-cel_silverstone-r0/plugins/sfputil.py

```diff
@@ -409,16 +409,32 @@ def _write_byte(self, port_num, devid, page, off, val):
             f.seek(addr)
             f.write(chr(val))
         except Exception as ex:
-            log_info("write failed: {0}".format(ex))
+            log_info("PORT {0}: pg={1}, ofs={2}: write failed: {3}".format(port_num, page, off, ex))
         finally:
             f.close()
 
-    def _init_cmis_module_custom(self, port_num, xcvr, hwsku):
-        # As of now, init sequence is only necessary for 'INNOLIGHT T-DP4CNT-N00'
-        if xcvr != 'INNOLIGHT T-DP4CNT-N00':
-            return True
-
-        log_info("PORT {0}: {1}: _init_cmis_module_custom".format(port_num, xcvr))
+    def _get_application_code(self, port_num, host_intf):
```

Comment:
There is no guarantee that 100G PAM4 mode is always the application 2, and hence we'll need to decode the application advertisement and figure the correct application code